WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -373,12 +373,21 @@ def update
       return
     end
     registration_attributes = registration_params
+    was_accepted = @registration.accepted?
+    was_deleted = @registration.deleted?
+    # The only case we go to this endpoint if the registration was deleted is when we regiter again.
+    if was_deleted
+      # Set the accepted_at/deleted_at to nil iff it's not already set,
+      # which can happen when moving from deleted to accepted.
+      registration_attributes = { accepted_at: nil, deleted_at: nil }.merge(registration_attributes)
+    end
+    # Don't rely on the status in the params, compute the new status from the
+    # timestamps.
+    new_status = Registration.new(registration_attributes.slice(:accepted_at, :deleted_at)).checked_status
```

Comment:
I get the point of doing this, but constructing a dummy database entity with only two parameters, and only using it to compute one status check feels very wrong.

Is there any chance of outsourcing this into a static/module method? Something along the lines of `Registration.compute_status(accepted_at, deleted_at)` (which could then also be re-used within the `Registration` class)

Reply:
Right it did feel weird to me to, and it didn't fully convince myself because there is no guarantee that in the future constructing a registration from just this would work.
I can try the static way but I'm afraid it will be ugly too; I'll check what it looks like.